packages/discovery-provider/src/tasks/create_engagement_notifications.py

```diff
@@ -0,0 +1,131 @@
+from datetime import datetime, timedelta
+
+from sqlalchemy import String, and_, func
+
+from src.models.notifications.notification import Notification
+from src.models.rewards.challenge import Challenge
+from src.models.rewards.challenge_disbursement import ChallengeDisbursement
+from src.models.rewards.user_challenge import UserChallenge
+from src.tasks.celery_app import celery
+from src.utils.structured_logger import StructuredLogger, log_duration
+
+logger = StructuredLogger(__name__)
+BATCH_SIZE = 500
+CLAIMABLE_REWARD = "claimble_reward"
+START_DATETIME = datetime(2024, 6, 6)
+
+
+def get_claimable_reward_notification_group_id(user_id, challenge_id, specifier):
+    return (
+        f"{CLAIMABLE_REWARD}:{user_id}:challenge:{challenge_id}:specifier:{specifier}"
+    )
+
+
+@log_duration(logger)
+def _create_engagement_notifications(session):
+    week_cooldown_interval = timedelta(weeks=1)
+
+    new_notifications = []
+
+    user_challenges_cooldown_completed = (
+        session.query(UserChallenge)
+        .outerjoin(
+            ChallengeDisbursement,
+            and_(
+                UserChallenge.challenge_id == ChallengeDisbursement.challenge_id,
+                UserChallenge.specifier == ChallengeDisbursement.specifier,
+            ),
+        )
+        .join(Challenge, Challenge.id == UserChallenge.challenge_id)
+        .outerjoin(
+            Notification,
+            and_(
+                Notification.specifier == UserChallenge.specifier,
+                Notification.group_id
+                == func.concat(
+                    CLAIMABLE_REWARD,
```

Comment:
checking we cant use the `get_claimable_reward_notifaction_group_id` here?

Reply:
i tried but it wouldn't work in this sqlalchemy query